montepy/data_inputs/data_input.py

```diff
@@ -52,17 +54,29 @@ class DataInputAbstract(MCNP_Object):
 
     _classifier_parser = ClassifierParser()
 
-    def __init__(self, input=None, fast_parse=False):
+    def __init__(
+        self,
+        input: Union[montepy.input_parser.mcnp_input.Input, str] = None,
```

Comment:
Throughout:

The `Union[montepy.input_parser.mcnp_input.Input, str]` is reused frequently.

Consider creating and reusing this type somewhere. For example:

```python
"""
montepy typing
"""
import typing
from montepy.input_parser.mcnp_input import Input

T_InitInput = typing.Union[Input, str]
```